### DISCUSSION TITLE
How does listener access control work?

### DISCUSSION THREAD

**Discussion by gjalring** [emqx/emqx#8000] [Q&A] [answered]
I have tried getting the SSL listener access control to work, but it still allows all IPs to connect. What am I doing wrong?
Have not been able to find answers to my problem from google. 

I in addition to the rule 1 that is uncommented below, I have also tried setting a rule 2 to deny all other. It seems straight forward to set up, but I have not been able to. Also tried cidr notation with 10.113.84.150/32

```
## The access control rules for the MQTT/SSL listener.
## Value: ACL Rule
listener.ssl.external.access.1 = allow 10.113.84.150
##listener.ssl.external.access.2 = deny all
```

**Comment by thalesmg** [comment, marked as answer]
Hi @gjalring !

Perhaps anonymous authn is enabled?  https://www.emqx.io/docs/en/v4.4/advanced/auth.html#anonymous-authentication

Also check your `authacl.acl_nomatch` config.  https://www.emqx.io/docs/en/v4.4/configuration/configuration.html#acl-nomatch

**Reply by gjalring** [reply]
I have in emqx.conf
```
allow_anonymous = false
acl_nomatch = deny
acl_deny_action = disconnect
```
I have also set up rules in acl.conf that seems to be somewhat working
```
{allow,` {ipaddr, "10.113.85.150"}, publish, ["Topic1/#", "Topic2/#"]}.
{deny, all}.
```

The acl rules still lets other clients to connect, but not subscribe or publish topics. I thought my settings would also disconnect them.

**Reply by thalesmg** [reply]
The ACL rules are checked after the connection is made, and [only control publish/subscribe permissions](https://docs.emqx.com/en/enterprise/v4.4/advanced/acl.html).

To prevent connections from unknown clients, you have to enable one of the authorization mechanisms.  One example is the built-in database authn: https://docs.emqx.com/en/enterprise/v4.4/modules/mnesia_authentication.html
